docs/adr/0023-paypal-support.md

```diff
@@ -0,0 +1,102 @@
+- Recurly
+
+## Decision Outcome
+
+Originally considered to be the 'fast-path', using PayPal subscriptions ended up being approximately the same engineering effort to implement as Stripe-driven invoice processing. Given additional user experience benefits of the Stripe-driven approach, it is now the recommend implemenation path.
+
+## Pros and Cons of the Options
+
+### Stripe-driven out-of-band invoice processing
+
+Stripe defaults subscriptions to billing automatically using a payment source on file with Stripe. Stripe also provides the option to have subscriptions that have a `collection_method` of `send_invoice`. In this model, Stripe can track subscriptions, and generate invoices that can be paid `out-of-band`. This allows Stripe to have an accurate record of charges, invoices, and subscriptions but how the invoice is considered 'paid' is up to FxA.
+
+PayPal has a feature called reference transactions that allow FxA to get a customers authorization to charge their PayPal account. This authorization allows for later payments to be charged against the PayPal account. We will track this reference ID in Stripe metadata for the customer and mark the subscription as `send_invoice`.
+
+FxA will implement additional logic to listen for invoice creation webhooks and switch their `auto_advance` off if the subscription has metadata indicating it should be paid via a PayPal reference ID. We will then check the invoice for the amount to charge and run the PayPal charge via PayPal API. The invoice will then be updated to reflect whether it was paid successfully or not.
+
+Pros:
+
+- Stripe continues to serve as an accurate source of truth regarding a customers subscription.
+- All the logic that determines active subscriptions and entitlements using Stripe can be used as-is.
+- Multiple product support works as-is.
+- Support representatives can see accurate payment history in Stripe for customers paying via PayPal.
+- Support can credit a PayPal customer, just as they can for a credit card customer.
+- Can extend in a similar manner to track a correlating IAP for a user.
+- Can handle switching payment method from credit card to PayPal reference ID.
+
+Cons:
+
+- Additional metadata in Stripe can start pushing us closer to metadata limits on objects.
+- Sensitive processing logic that runs PayPal transactions which we need to ensure work correctly.
+- We still could end up storing PayPal reference ID's in a separate db table to ensure we track
+  prior PayPal billing.
+
+### PayPal Subscriptions linked/tracked by FxA
+
+In this approach, we would create products/plans in PayPal, and link them back to the Stripe equivilant by updating Stripe metadata on the products/plans. FxA would also gain an additional database table that maps PayPal subscription id's to the Fx account. All the logic in FxA that looks up a users subscriptions would be updated to check the PayPal linking table to determine what subscriptions the user has active.
+
```

Comment:
Small typo on "equivalent".

Reply:
"ids" instead of "id's" but "user's" instead of "users".

And to nitpick, I find "what active subscriptions the user has" reads a little better.